apps/auth-server/routes/routes.js

```diff
@@ -105,6 +105,30 @@ const addCsrfGlobal = (req, res, next) => {
     next();
 };
 
+const rateLimits = {};
+const AUTH_LIMIT = 5000;
+const AUTH_WINDOW_MS = 60 * 1000;
+
+const rateLimiter = (req, res, next) => {
+    const ip = req.ip;
+    if (!rateLimits[ip]) {
+        rateLimits[ip] = { count: 1, startTime: Date.now() };
+    } else {
+        const currentTime = Date.now();
+        if (currentTime - rateLimits[ip].startTime < AUTH_WINDOW_MS) {
+            rateLimits[ip].count++;
+        } else {
+            rateLimits[ip] = { count: 1, startTime: currentTime };
+        }
+    }
+
+    if (rateLimits[ip].count > AUTH_LIMIT) {
+        return res.status(429).send('Too many requests. Please try again later.');
+    }
+
+    next();
+};
+
```

Comment:
This method looks a lot like the other `rateLimiter.js`. Can you refactor this to a shared package?